Frameworks/UIKit/UIColor.mm

```diff
@@ -33,790 +33,479 @@
-        TraceVerbose(TAG, L"Selecting pattern %hs", pPattern);
-
-        return [[[self class] performSelector:NSSelectorFromString(pattern)] retain];
-    } else {
-        if ([coder containsValueForKey:@"UIWhite"]) {
-            const CGFloat uiWhite = [coder decodeFloatForKey:@"UIWhite"];
-            _components.r = uiWhite;
-            _components.g = uiWhite;
-            _components.b = uiWhite;
-        } else {
-            _components.r = [coder decodeFloatForKey:@"UIRed"];
-            _components.g = [coder decodeFloatForKey:@"UIGreen"];
-            _components.b = [coder decodeFloatForKey:@"UIBlue"];
-        }
-        if ([coder containsValueForKey:@"UIAlpha"]) {
-            _components.a = [coder decodeFloatForKey:@"UIAlpha"];
-        } else {
-            _components.a = 1.0f;
-        }
-
-        return self;
-    }
++ (UIColor*)colorWithHue:(float)h saturation:(float)s brightness:(float)v alpha:(float)a {
+    return [[[self alloc] initWithHue:h saturation:s brightness:v alpha:a] autorelease];
 }
 
 /**
  @Status Interoperable
 */
-- (void)encodeWithCoder:(NSCoder*)coder {
-    assert(_type == solidBrush);
-
-    [coder encodeFloat:_components.r forKey:@"UIRed"];
-    [coder encodeFloat:_components.g forKey:@"UIGreen"];
-    [coder encodeFloat:_components.b forKey:@"UIBlue"];
-    [coder encodeFloat:_components.a forKey:@"UIAlpha"];
++ (UIColor*)colorWithWhite:(float)gray alpha:(float)alpha {
+    return [[[self alloc] initWithWhite:gray alpha:alpha] autorelease];
 }
 
 /**
  @Status Interoperable
 */
-+ (UIColor*)grayColor {
-    static UIColor* color = [_UICachedColor colorWithRed:0.5f green:0.5f blue:0.5f alpha:1.0f];
-    return color;
++ (UIColor*)colorWithPatternImage:(UIImage*)image {
+    return [[[self alloc] initWithPatternImage:image] autorelease];
 }
 
 /**
  @Status Interoperable
 */
-+ (UIColor*)cyanColor {
-    static UIColor* color = [_UICachedColor colorWithRed:0.0f green:1.0f blue:1.0f alpha:1.0f];
-    return color;
++ (UIColor*)colorWithCGColor:(CGColorRef)clr {
+    return [[[self alloc] initWithCGColor:clr] autorelease];
 }
 
 /**
  @Status Interoperable
 */
-+ (UIColor*)magentaColor {
-    static UIColor* color = [_UICachedColor colorWithRed:1.0f green:0.0f blue:1.0f alpha:1.0f];
-    return color;
+- (CGColorRef)CGColor {
+    return NSInvalidAbstractInvocationReturn();
 }
 
 /**
- @Status Interoperable
+ @Status Stub
 */
-+ (UIColor*)lightGrayColor {
-    static UIColor* color = [_UICachedColor colorWithRed:0.75f green:0.75f blue:0.75f alpha:1.0f];
-    return color;
+- (instancetype)initWithCGColor:(CGColorRef)cgColor {
+    return NSInvalidAbstractInvocationReturn();
 }
 
 /**
  @Status Interoperable
 */
-+ (UIColor*)darkGrayColor {
-    static UIColor* color = [_UICachedColor colorWithRed:0.25f green:0.25f blue:0.25f alpha:1.0f];
-    return color;
+- (instancetype)initWithHue:(float)h saturation:(float)s brightness:(float)v alpha:(float)a {
+    return NSInvalidAbstractInvocationReturn();
```

Comment:
yes, it's our own implementation. 

---
In reply to: [113287177](https://github.com/Microsoft/WinObjC/pull/2538#discussion_r113287177) [](ancestors = 113287177)